src/Domain/Model/Donation.php

```diff
@@ -149,9 +149,25 @@ public function confirmBooked() {
 			throw new RuntimeException( 'Only incomplete donations can be confirmed as booked' );
 		}
 
+		if ( $this->hasComment() && ( $this->needsModeration() || $this->isCancelled() ) ) {
+			$this->makeCommentPrivate();
+		}
+
 		$this->status = self::STATUS_EXTERNAL_BOOKED;
 	}
 
+	private function makeCommentPrivate() {
+		$this->comment = new DonationComment(
+			$this->comment->getCommentText(),
+			false,
```

Comment:
Maybe. The context here is very small and specific though. Anyway, feel free to make PR.

Another possibility for this code is to move it into a `createPrivateCopy` method in `DonationComment`.